File: QytetetRuby/lib/sorpresa.rb
#encoding: utf-8
# Francisco Beltrán Sánchez
# Pablo Cordero Romero
module ModeloQytetet
  
  class Sorpresa
    #Consultores de forma implícita
    attr_reader :texto
    attr_reader :valor
    attr_reader :tipo

    def initialize(tx, vl, tp)
      @texto = tx
      @valor = vl
      @tipo = tp
    end

    def to_s
     "Texto: #{@texto} \n Valor: #{@valor} \n Tipo: #{@tipo}"
    end

=begin
    Así serían los consultores de forma explicita:

    def texto
      @texto
    end

    def valor
      @valor
    end

    def tipo
     @tipo
    end
=end

  end
end

File: QytetetJava/src/modeloqytetet/Sorpresa.java
/**
 * Practica de:
 * 
 * Francisco Beltrán Sánchez
 * Pablo Cordero Romero
 */
package modeloqytetet;

public class Sorpresa {
    private String texto;
    private TipoSorpresa tipo;
    private int valor;
    
    Sorpresa(String tx, int vl, TipoSorpresa tp){
        this.texto = tx;
        this.tipo = tp;
        this.valor = vl;
    }
    
    String getTexto(){
        return texto;
    }
    
    TipoSorpresa getTipo(){
        return tipo;
    }
    
    int getValor(){
        return valor;
    }
    
    @Override
    public String toString() {
        return "\n-----\nSorpresa:\n" + "\ttexto=" + texto + "\n\tvalor=" + Integer.toString(valor) + "\n\ttipo=" + tipo;
    }

}
